andela/balder-ah-backend
# #162170721 Users should be able to retrieve all profiles

#### What does this PR do?
This PR handles retrieving a list of all user's profiles

#### Description of Task to be completed?
Have the following endpoint working
GET /api/profiles
- Write tests for this  endpoint
- Create controller for retrieving all user profiles except logged in user
- User should see a list of author's profiles on the platform with the number of articles they have
- Document this API with Swagger API Doc

#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a GET request to **localhost:3000/api/profiles** to retrieve all user's profiles
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**

#### Any background context you want to provide?
- Authentication is required
- Only an authenticated user can the list of all the profiles

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170721](https://www.pivotaltracker.com/story/show/162170721)


#### Screenshots (If applicable)

<img width="1440" alt="screen shot 2018-12-14 at 5 04 02 am" src="https://user-images.githubusercontent.com/29441898/49983243-d1b4d580-ff61-11e8-826c-cb1cce3aea78.png">


#### Questions:
